### PR TITLE
fix: allow removal of a tab when there is no selectedTab

### LINKED ISSUE #4608: Vaadin 24: Can't remove a Tab if there is no selectedTab
### Description

If there is no selected tab (=null) there is an exception when you try to remove an existing tab.

> java.lang.IllegalArgumentException: The 'index' argument should be greater than or equal to 0. It was: -1

### Expected outcome

No exception and the tab should be removed.

### Minimal reproducible example

```
public class HelloWorldView extends HorizontalLayout {

    public HelloWorldView() {
        Tab details = new Tab("Details");
        Tab payment = new Tab("Payment");
        Tab shipping = new Tab("Shipping");

        Tabs tabs = new Tabs(details, payment, shipping);

        add(tabs);
        tabs.setSelectedTab(null);
        tabs.remove(details);
    }

}

```

### Steps to reproduce

Just run this view.

The problem has been introduced here https://github.com/vaadin/flow-components/commit/b9597239e12b52984183801664b0680d1b53eb38#diff-2dae6bea09cab6bd2fd6407f25a551fa8a15b8c080f5069926d82c8b2bdece74R240

### Environment

Vaadin version(s): 24.0.0.alpha9

### Browsers

_No response_

### PR DESCRIPTION
**Pull request by jcgueriaud1**
## Description

Please list all relevant dependencies in this section and provide summary of the change, motivation and context.

Fixes #4608 

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/contributing/overview
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.
- [ ] I have not completed some of the steps above and my pull request can be closed immediately.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

**Review by web-padawan — COMMENTED** [review]
@jcgueriaud1 There's a checkbox that you apparently checked without reading through 🤓 

>  I have not completed some of the steps above and my pull request can be closed immediately.

**Comment by jcgueriaud1** [comment]
Technically I didn't check it, it's prefilled and said in a negative way. That's not GDPR compliant :) 
It's updated.

**Comment by web-padawan** [comment]
> Technically I didn't check it, it's prefilled and said in a negative way. That's not GDPR compliant

Well, actually it was left that way intentionally as an attempt to draw someon's attention 😅

File: vaadin-tabs-flow-parent/vaadin-tabs-flow/src/main/java/com/vaadin/flow/component/tabs/Tabs.java [hunk 1]
```diff
@@ -237,7 +237,7 @@ public void remove(Tab... tabs) {
         int lowerIndices = (int) Stream.of(tabs).map(this::indexOf)
                 .filter(index -> index >= 0 && index < selectedIndex).count();
 
-        Tab selectedTab = getTabAt(selectedIndex);
+        Tab selectedTab = (selectedIndex < 0) ? null : getTabAt(selectedIndex);
```
Comment (sissbruecker) [inline vaadin-tabs-flow-parent/vaadin-tabs-flow/src/main/java/com/vaadin/flow/component/tabs/Tabs.java:240]: Could just use `getSelectedTab` instead.

Comment (jcgueriaud1) [reply vaadin-tabs-flow-parent/vaadin-tabs-flow/src/main/java/com/vaadin/flow/component/tabs/Tabs.java]: True, updated.

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**